code/game/machinery/doors/door.dm

```diff
@@ -502,3 +502,10 @@
 /obj/machinery/door/zap_act(power, zap_flags)
 	zap_flags &= ~ZAP_OBJ_DAMAGE
 	. = ..()
+
+/obj/machinery/door/CanPathfindPass(obj/item/card/id/ID, to_dir, atom/movable/caller, no_id)
+	if(caller?.checkpass(PASSDOOR) && !locked)
+		return TRUE
+	if(caller.checkpass(PASSGLASS))
```

Comment:
Not necessary to add ```caller?.``` here too?

Reply:
good point, I just copied this from the canpass